ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java

```diff
@@ -195,6 +196,23 @@ public void shouldHandleMultiDimensionalArray() throws Exception {
         assertThat(result, equalTo("item_11"));
     }
 
+    @Test
+    public void shouldHandleMultiDimensionalArrayWithBaseStartingFrom1() throws Exception {
+        final KsqlConfig localKsqlConfig =
+            new KsqlConfig(Collections.singletonMap(KsqlConfig.KSQL_FUNCTIONS_ARRAY_LEGACY_BASE_CONFIG, false));
+        final CodeGenRunner localCodeGenRunner = new CodeGenRunner(schema, localKsqlConfig, functionRegistry);
```

Comment:
consider making `ksqlConfig` field a mock, which means you no longer need to create a new running, just set up the mock.